components/content_settings/core/browser/brave_cookie_settings.cc

```diff
@@ -40,60 +38,34 @@ void BraveCookieSettings::GetCookieSetting(const GURL& url,
   DCHECK(cookie_setting);
   CookieSettings::GetCookieSetting(url, first_party_url, source,
       cookie_setting);
-
   if (*cookie_setting == CONTENT_SETTING_BLOCK) {
-    *cookie_setting = CONTENT_SETTING_BLOCK;
     return;
   }
 
-  GURL primary_brave_url = tab_url == GURL("about:blank") ?
-      first_party_url : tab_url;
+  GURL primary_url = (tab_url == GURL("about:blank") || tab_url.is_empty() ?
+      first_party_url : tab_url);
 
-  // Check the Brave shields setting, if it is off, just return without
-  // blocking anything.
   ContentSetting brave_shields_setting =
       host_content_settings_map_->GetContentSetting(
-          primary_brave_url, GURL(),
+          primary_url, GURL(),
           CONTENT_SETTINGS_TYPE_PLUGINS, brave_shields::kBraveShields);
-  if (brave_shields_setting == CONTENT_SETTING_BLOCK) {
-    *cookie_setting = CONTENT_SETTING_ALLOW;
-    return;
-  }
-
-  // First party setting of block means always block everything
   ContentSetting brave_1p_setting = host_content_settings_map_->GetContentSetting(
-      primary_brave_url, GURL("https://firstParty/"),
+      primary_url, GURL("https://firstParty/"),
       CONTENT_SETTINGS_TYPE_PLUGINS, brave_shields::kCookies);
-  if (brave_1p_setting == CONTENT_SETTING_BLOCK) {
-    *cookie_setting = CONTENT_SETTING_BLOCK;
-    return;
-  } else if (brave_1p_setting == CONTENT_SETTING_DEFAULT) {
-    // in absence of an explicit rule, allow first-party cookies by default
-    brave_1p_setting = CONTENT_SETTING_ALLOW;
-  }
-
-  // Third party setting of allow, means always allow everything
   ContentSetting brave_3p_setting =
       host_content_settings_map_->GetContentSetting(
-          primary_brave_url, GURL(),
+          primary_url, GURL(),
           CONTENT_SETTINGS_TYPE_PLUGINS, brave_shields::kCookies);
-  if (brave_3p_setting == CONTENT_SETTING_ALLOW) {
-    *cookie_setting = CONTENT_SETTING_ALLOW;
-    return;
-  } else if (brave_3p_setting == CONTENT_SETTING_DEFAULT) {
-    // in absence of an explicit rule, block third-party cookies by default
-    brave_3p_setting = CONTENT_SETTING_BLOCK;
-  }
 
-  // Otherwise determine based on if the cookie is third-party or not.
-  // Empty first-party URL indicates a first-party request, so use the
-  // previously obtained cookie_setting in that case.
-  if (!first_party_url.is_empty()) {
-    if (SameDomainOrHost(url, first_party_url, INCLUDE_PRIVATE_REGISTRIES)) {
-      *cookie_setting = brave_1p_setting;
-    } else {
-      *cookie_setting = brave_3p_setting;
-    }
+  bool allow_brave_shields = brave_shields_setting == CONTENT_SETTING_ALLOW ||
+    brave_shields_setting == CONTENT_SETTING_DEFAULT;
+  bool allow_1p_cookies = brave_1p_setting == CONTENT_SETTING_ALLOW ||
+    brave_1p_setting == CONTENT_SETTING_DEFAULT;
+  bool allow_3p_cookies = brave_3p_setting == CONTENT_SETTING_ALLOW;
+
+  if (ShouldBlockCookie(allow_brave_shields, allow_1p_cookies,
```

Comment:
I was hoping to get rid of this whole file, but it turns out it's still needed even with the network delegate bevause of local storage and indexdb which uses it.